File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size() / 2;
        sort(nums.begin(), nums.end());
        int cnt;
        int pt = pow(10,9) + 1;
        for(int i = 0; i < nums.size(); i++){
            if(pt != nums[i]){
                if(cnt > n) return pt;
                pt = nums[i];
                cnt = 1;
            }else{
                cnt++;
            }
        }
        return pt;
    }
};

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums) {
        //  풀이 방법
        //  https://leetcode.com/problems/majority-element/solutions/3133635/c-solution/

        //  보이어-무르 과반수 투표 알고리즘
        //  https://sgc109.github.io/2020/11/30/boyer-moore-majority-vote-algorithm/
        int count = 0;
        int? candidate = null;

        foreach(int num in nums){
            if(count == 0) candidate = num;
            count += (num == candidate) ? 1 : -1;
            Console.WriteLine(count);
        }

        return candidate.Value;
    }
}